hal/src/rtl872x/usart_hal.cpp

```diff
@@ -179,7 +179,15 @@ class Usart {
         }
         PAD_PullCtrl(hal_pin_to_rtl_pin(txPin_), GPIO_PuPd_UP);
         PAD_PullCtrl(hal_pin_to_rtl_pin(rxPin_), GPIO_PuPd_UP);
-
+        // Configure CTS/RTS pins
+        if (ctsPin_ != PIN_INVALID) {
+            Pinmux_Config(hal_pin_to_rtl_pin(ctsPin_), PINMUX_FUNCTION_UART_RTSCTS);
+            PAD_PullCtrl(hal_pin_to_rtl_pin(ctsPin_), GPIO_PuPd_UP);
+        }
+        if (rtsPin_ != PIN_INVALID) {
+            Pinmux_Config(hal_pin_to_rtl_pin(rtsPin_), PINMUX_FUNCTION_UART_RTSCTS);
+            PAD_PullCtrl(hal_pin_to_rtl_pin(rtsPin_), GPIO_PuPd_UP);
+        }
```

Comment:
This change was needed for the UART hardware flow control to work as expected.

Reply:
cc @XuGuohui